lib/spec/staticcacher.sw-spec.js

```diff
@@ -0,0 +1,181 @@
+
+      return defaultStaticCacher.getDefaultCache().then(function(cache) {
+        cacheGot = cache;
+        return caches.open(defaultCacheName)
+      }).then(function(expectedCache) {
+        expect(cacheGot).to.equal(expectedCache);
+      });
+    });
+  });
+
+  describe('addAll()', function() {
+    afterEach(function() {
+      defaultStaticCacher.fetchAndCache.restore();
+    });
+
+    it('should work with polyfill', function() {
+      sinon.stub(defaultStaticCacher, 'fetchAndCache', function(request, cache) {
+        var response = new Response(request.url, {
+          status: 200
+        });
+        return cache.put(request.clone(), response.clone());
+      });
+
+      var defaultCache;
+
+      return defaultStaticCacher.getDefaultCache().then(function(cache) {
+        defaultCache = cache;
+
+        cache.addAll = undefined;
+        return defaultStaticCacher.addAll(cache, ['http://www.example.com/',
+          'http://www.mozilla.org/'
+        ]);
+      }).then(function() {
+        expect(defaultCache).to.be.ok;
+        sinon.assert.calledTwice(defaultStaticCacher.fetchAndCache);
+        return defaultCache.match(new Request('http://www.example.com/'));
+      }).then(function(response) {
+        return response.text();
+      }).then(function(text) {
+        expect(text).to.equal('http://www.example.com/');
+      });
+    });
+  });
+
+  describe('fetchAndCache()', function() {
+    var content = 'my content';
+
+    beforeEach(function() {
+      self.fetch.restore();
+    })
+
+    it('should fetch and cache url a valid resource', function() {
+      sinon.stub(self, 'fetch', function() {
+        var response = new Response(content, {});
+        return Promise.resolve(response);
+      });
+
+      var defaultCache;
+
+      return defaultStaticCacher.getDefaultCache().then(function(cache) {
+        defaultCache = cache;
+      }).then(function() {
+        return defaultStaticCacher.fetchAndCache(new Request(defaultUrls[0]), defaultCache);
+      }).then(function() {
+        return defaultCache.match(new Request(defaultUrls[0]));
+      }).then(function(response) {
+        return response.text();
+      }).then(function(text) {
+        expect(text).to.be.equal(content);
+      });
+    });
+
+    it('should not cache invalid resource', function() {
+      sinon.stub(self, 'fetch', function() {
+        var response = new Response('invalid content', {
+          status: 401
+        });
+        return Promise.resolve(response);
+      });
+
+      var defaultCache;
+
+      return defaultStaticCacher.getDefaultCache().then(function(cache) {
+        defaultCache = cache;
+        return defaultStaticCacher.fetchAndCache(new Request(defaultUrls[0]), defaultCache);
+      }).then(function() {
+        sinon.assert.calledOnce(self.fetch);
+        return defaultCache.match(new Request(defaultUrls[0]));
+      }).then(function(response) {
+        // Shouldn't we catch instead of resolving undefined?
```

Comment:
That case is actually covered [by the specs](www.w3.org/TR/service-workers/#cache-match). I'll remove that comment before merging the PR